airflow/dag_processing/processor.py

```diff
@@ -763,8 +763,29 @@ def _execute_dag_callbacks(self, dagbag: DagBag, request: DagCallbackRequest, se
         if callbacks and context:
             DAG.execute_callback(callbacks, context, dag.dag_id)
 
-    def _execute_task_callbacks(self, dagbag: DagBag | None, request: TaskCallbackRequest, session: Session):
-        if not request.is_failure_callback:
+    def _execute_task_callbacks(
+        self, dagbag: DagBag | None, request: TaskCallbackRequest, session: Session
+    ) -> None:
+        """
+        Execute the task callbacks.
+
+        :param dagbag: the DagBag to use to get the task instance
+        :param request: the task callback request
+        :param session: the session to use
+        """
+        try:
+            callback_type = TaskInstanceState(request.task_callback_type)
```

Comment:
Why do we need to type cast here for request.task_callback_type? It seems that task_callback_type is either TaskInstanceState or None, or am I missing something?

Reply:
I believe the `TaskInstanceState` enum class returns all valid task instance states. If it's not valid, it will result in an error, and the state here should be none, which is handled in the exception part.